lib/cisco_node_utils/client/nxapi/client.rb

```diff
@@ -103,59 +102,56 @@ def cache_flush
     }
   end
 
-  # Configure the given command(s) on the device.
-  #
-  # @raise [CliError] if any command is rejected by the device
-  #
-  # @param commands [String, Array<String>] either of:
-  #   1) The configuration sequence, as a newline-separated string
-  #   2) An array of command strings (one command per string, no newlines)
-  def config(commands)
-    super
-    if commands.is_a?(String)
-      commands = commands.split(/\n/)
-    elsif !commands.is_a?(Array)
-      fail TypeError
-    end
-    req('cli_conf', commands)
-  end
-
-  # Executes a command in exec mode on the device.
-  #
-  # If cache_auto? (on by default) is set then the CLI cache will be flushed.
+  # Configure the given CLI command(s) on the device.
   #
-  # For "show" commands please use show() instead of exec().
+  # @raise [RequestNotSupported] if this client doesn't support CLI config
   #
-  # @param command [String] the exec command to execute
-  # @return [String, nil] the body of the output of the exec command
-  #   (if any)
-  def exec(command)
+  # @param data_format one of Cisco::DATA_FORMATS. Default is :cli
+  # @param context [String, Array<String>] Zero or more configuration commands
+  #   used to enter the desired CLI sub-mode
+  # @param values [String, Array<String>] One or more commands
+  #   to enter within the CLI sub-mode.
+  def set(data_format: :cli,
```

Comment:
I could certainly add data-model-specific functions (`cli_set(**args) == set(data_format: :cli, **args)`) but I'm not sure if there's a point. The goal is to make this as transparent as possible to the Node - it just gets stuff from the CmdRef and passes it through to the Client.